### ISSUE TITLE
Improve integration of Linux tools with Windows command line

### ISSUE THREAD

**Issue by rustyx** [Microsoft/BashOnWindows#1823]
It seems to be impossible to execute Linux tools from `cmd` or anywhere else outside bash.exe.

For example, I can't run `grep` from `cmd`:
```
C:\>C:\Users\test\AppData\Local\lxss\rootfs\bin\grep
'C:\Users\test\AppData\Local\lxss\rootfs\bin\grep' is not recognized as an internal or 
external command, operable program or batch file.

C:\>C:\Users\test\AppData\Local\lxss\rootfs\bin\grep.
'C:\Users\test\AppData\Local\lxss\rootfs\bin\grep.' is not recognized as an internal or 
external command, operable program or batch file.
```

If I give `grep` the `.exe` extension, it still doesn't work:
```
C:\>bash -c "sudo cp /bin/grep /bin/grep.exe"

C:\>C:\Users\test\AppData\Local\lxss\rootfs\bin\grep
This version of C:\Users\me\AppData\Local\lxss\rootfs\bin\grep.exe is not compatible with
the version of Windows you're running. Check your computer's system information and then
contact the software publisher.
```

It would have been **great** to be able to use Linux tools directly from Windows scripts. Otherwise I am kind of forced to keep cyg \*cough\* win installed.

Maybe automatically give executable ELF files an .exe extension and/or have the Windows subsystem recognize them and forward to Lxss?

**Comment by jackchammons** [comment]
Pretty interesting idea. We'll keep this in mind for our future planning.

**Comment by therealkenc** [comment]
At one point there was this [User Voice](https://wpdev.uservoice.com/forums/266908-command-prompt-console-bash-on-ubuntu-on-windo/suggestions/13292958-rethink-your-strategy-on-how-the-linux-bash-integr) entry, but it was locked.

**Comment by sunilmut** [comment]
I agree that there are merit in this post and the User Voice ticket (thanks @therealkenc) for it to be considered. @bitcrazed to see if we can reopen it, so that people can continue to vote and discuss. We can always add comment to User Voice ticket stating that this is not being considered at the moment. But, there is little value in closing a ticket, when it is not complete.

**Comment by therealkenc** [comment]
Appreciate the gesture, but understand at this point the issue is pretty buried over there. We can discuss WSL bumping `__NR_read` up from `0x0000` to `0x1000` so it doesn't conflict with `NtAccessCheck` in 2018 or 2020. 🐱‍👤

**Comment by mhamrah** [comment]
Just want to +1 this. Would love to be able to run executables located within WSL from windows without having to specify 'wsl' as the command. For instance, if I have something like `flow`, or `go` in my `PATH` in WSL, that is executed in WSL, I'd love to call `flow` or `go` directly from the Windows side. This will make running windows apps, like VS code, whose extensions require a specific set of commands available on the PATH, easier. Currently I see engineers setting up some x-terminal so VS Code runs under linux and presented via the x-windows system.  By making executables located within WSL available natively on the windows side--without having to run them through the wsl command--will make the wsl a very seamless experience from Windows.
